pilot/pkg/security/authz/model/generator.go

```diff
@@ -145,39 +147,67 @@ func (srcPrincipalGenerator) principal(key, value string, _ bool, useAuthenticat
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-type requestPrincipalGenerator struct{}
+type requestPrincipalGenerator struct {
+	useExtendedJwt bool
+}
 
 func (requestPrincipalGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
 	return nil, fmt.Errorf("unimplemented")
 }
 
-func (requestPrincipalGenerator) principal(key, value string, forTCP bool, _ bool) (*rbacpb.Principal, error) {
+func (rpg requestPrincipalGenerator) principal(key, value string, forTCP bool, _ bool) (*rbacpb.Principal, error) {
 	if forTCP {
 		return nil, fmt.Errorf("%q is HTTP only", key)
 	}
-
+	if rpg.useExtendedJwt {
+		iss, sub, _ := strings.Cut(value, "/")
+		if value == "*" {
+			iss, sub = "*", "*"
+		}
+		im := MetadataMatcherForJWTClaims([]string{"iss"}, matcher.StringMatcher(iss), true)
+		sm := MetadataMatcherForJWTClaims([]string{"sub"}, matcher.StringMatcher(sub), true)
+		return principalAnd([]*rbacpb.Principal{principalMetadata(im), principalMetadata(sm)}), nil
```

Comment:
I think we would now allow matching `a*/b` where before we would not since we treated the whole thing as one unit.

Which is probably reasonable. Not sure why we didn't just make 2 fields to begin with

Reply:
Yeap, added a fix for this.